core/src/main/java/org/infinispan/security/impl/SubjectACL.java

```diff
@@ -0,0 +1,17 @@
+package org.infinispan.security.impl;
+
+import java.util.Set;
+
+/**
+ * SubjectRoles.
+ *
+ * @author Tristan Tarrant
+ * @since 8.1
+ */
+public class SubjectACL {
```

Comment:
Some comments:
* it looks like it is stored in cache. An externalizer would be fine
* the ACL mask could be calculated in the class and cached (it does not changes right?)
* the roles should not be exposed and a containsRole method should be added
* the roles could be stored as a sorted array and use binarySearch to find it (assuming that a subject as small role set, it would be memory efficient)